File: app/src/main/java/com/dubedivine/samples/data/model/Tag.java
package com.dubedivine.samples.data.model;

import kotlin.jvm.internal.Intrinsics;
import org.springframework.data.annotation.Id;
import org.springframework.data.mongodb.core.index.Indexed;
import org.springframework.data.mongodb.core.mapping.Document;
import za.co.dubedivine.networks.model.shared.TagBase;

import java.util.Date;
import java.util.Set;

/**
 * Created by divine on 2017/08/13.
 */


//@QueryEntity what is this
@Document
public class Tag extends TagBase {

    public Tag() {
    }

    public Tag(String name) {
        super(name);
    }
}

File: app/src/main/java/com/dubedivine/samples/data/model/Tag.kt
package com.dubedivine.samples.data.model

import java.io.Serializable
import java.util.*

/**
 * Created by divine on 2017/08/13.
 */


class Tag(// name is the ID means
        val name: String,
        val createAt: Date) : Serializable {
   ;
    private var questions: MutableList<Question>? = null


    fun getQuestions(): List<Question>? {
        return questions
    }

    fun setQuestions(questions: MutableList<Question>) {
        this.questions = questions
    }

    fun addQuestion(q: Question) {
        if (questions != null) {
            questions!!.add(q)
        } else {
            questions = ArrayList()
            questions!!.add(q)
        }
    }
}
